phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -24,14 +24,19 @@
 import static org.apache.phoenix.monitoring.GlobalClientMetrics.GLOBAL_MUTATION_COMMIT_TIME;
 import static org.apache.phoenix.query.QueryServices.WILDCARD_QUERY_DYNAMIC_COLS_ATTRIB;
 import static org.apache.phoenix.query.QueryServicesOptions.DEFAULT_WILDCARD_QUERY_DYNAMIC_COLS_ATTRIB;
+import static org.apache.phoenix.schema.types.PDataType.FALSE_BYTES;
+import static org.apache.phoenix.schema.types.PDataType.TRUE_BYTES;
```

Comment:
There is a MutationStateIT class, it might count as unit/it tests. 
My changes are mostly refactors. There was a huge send() method, I refactored and moved some part of send code into sendMutations function. This is the place where client creates and sends mutations for index and data table.